doc/pull-requests.md

```diff
@@ -0,0 +1,155 @@
+
+### Requesting Changes
+
+<a name="changes-review-reqs"></a>
+- [1.](#changes-review-reqs) Changes should be requested if an author does not follow [review requirements](#review-reqs).
+<a name="changes-no-nit"></a>
+- [2.](#changes-no-nit) Changes cannot be requested because of the personal preferences of a reviewer.
+  - Such change requests should be dismissed.
+<a name="changes-no-out-of-scope"></a>
+- [3.](#changes-no-out-of-scope) Changes cannot be requested if they address issues out of the scope of a PR.
+  - Such change requests should be dismissed and an issue should be filed to address them separately.
+<a name="changes-style-agreement"></a>
+- [4.](#changes-style-agreement) Styles and coding preferences should not be discussed on the PR, but raised to [the dev meeting](https://github.com/theia-ide/theia/wiki/Dev-Meetings),
+  agreed by the team, applied to [the coding guidelines](https://github.com/theia-ide/theia/wiki/Coding-Guidelines) and after that followed by all contributors.
+
+### Approving
+
+<a name="justifiying-approve"></a>
+- [1.](#justifiying-approve) Each approve should have supporting comments following these guidelines.
+<a name="dismissing-approve"></a>
+- [2.](#dismissing-approve) An approve without a comment should be dismissed.
+
+### Collaborating
+
+<a name="collaboration-on-pr"></a>
+- [1.](#collaboration-on-pr) If a change request is important, but cannot be elaborated by a reviewer,
+then a reviewer should be encouraged to open an alternative PR or collaborate on a current PR.
+<a name="completing-pr"></a>
+- [2.](#completing-pr) If a PR is important, but an author cannot or does not want to address outstanding issues,
+then maintainers can complete the PR with additional commits
+given that author commits are preserved, [signed-off](https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#sign-your-work) and an author accepted the [ECA](https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#eclipse-contributor-agreement).
+<a name="suggesting-help-on-pr"></a>
+- [3.](#suggesting-help-on-pr) Reviewers have to suggest his help via a comment to avoid intervening in an author work.
+<a name="landing-stale-pr"></a>
+- [4.](#landing-stale-pr) Such comment is not required if an author is not responsive.
+
+## Landing
+
+<a name="landing-pr"></a>
+- [1.](#landing-pr) A PR can be landed when:
+  - CI build is succeeded.
+  - An author accepted the [Eclipse Contributor Agreement](https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#eclipse-contributor-agreement).
+  - All checks from [the review checklist](#pull-request-review-checklist) are approved at least by one reviewer.
+  - There are no unresolved review comments.
+
+## Reverting
+
+<a name="reverting-pr"></a>
+- [1.](#reverting-pr) If a PR causes regressions after landing
+then an author has 2 days to resolve them after that a PR has to be reverted.
```

Comment:
> has 2 days

How will this be managed?

Reply:
We will ask an author to look into a fix if there is no response and we don't have time to fix it ourself then we revert it. As it was done here: https://github.com/theia-ide/theia/pull/5770

Reply:
Also if an author is responsive and trying to fix, but we keep receiving bug reports from users as it was the case with SCM PR, then we will revert everything and wait for the better PR.

Reply:
Do you think we need to rephrase it somehow to make it more clear?